### PR TITLE
fix: support pybind11 2.10

### PR DESCRIPTION
**Pull request by henryiii**

### CODE REVIEW

**Review by Skylion007 — APPROVED** [review]

**Comment by henryiii** [comment]
To be clear, this is just showing that it's broken. Don't know what the fix is yet. :)

**Comment by henryiii** [comment]
This has stopped picking up PyPy and is getting CPython instead:  `adding 'scikit_build_example/_core.cpython-37m-x86_64-linux-gnu.so'`.

File: CMakeLists.txt [hunk 1]
```diff
@@ -14,6 +14,8 @@ if(SKBUILD)
   list(APPEND CMAKE_PREFIX_PATH "${_tmp_dir}")
 endif()
 
+set(PythonLibsNew_FIND_VERSION " ")
```
Comment (henryiii) [inline CMakeLists.txt:17]:
```suggestion
set(PythonLibsNew_FIND_VERSION "3.6")
```

**Comment by henryiii** [comment]
Checking to see if setting `PythonLibsNew_FIND_VERSION` in pybind11 2.10 was a problem when looking for PyPy. Couldn't mask it out so just setting it with pybind11 2.9 instead.

**Comment by henryiii** [comment]
Nope, that wasn’t it

**Comment by henryiii** [comment]
Oh, wait, it might not have this var. Will try something else later.

File: pyproject.toml [hunk 2]
```diff
@@ -14,3 +14,7 @@ profile = "black"
 test-command = "pytest {project}/tests"
 test-extras = ["test"]
 test-skip = ["*universal2:arm64"]
+build-verbosity = 1
+linux.before-build = [
+    "rm -r _skbuild || echo "No previous run to clean",
```
Comment (henryiii) [inline pyproject.toml:19]:
```suggestion
    "rm -r _skbuild || echo 'No previous run to clean',
```

**Comment by henryiii** [comment]
Okay, I know what's wrong. The problem is the cache dir `_skbuild` is being persisted between runs. Now that pybind11 is respecting the "old" cache value for these settings, it's getting the last non-PyPy values filled in when it starts building for PyPy. This is the first place we notice a break (though it could happen elsewhere, values from a previous run are not always going to be "safe" to use!)

The core problem is we can't tell if someone passed `-D...`, in which case we want to respect it, and if it is just cached from a previous run. A few ideas for fixes:

* We delete the cached directory between cibuildwheel runs. I did that for linux above. Has to be done by every user, though.
* We have an opt-in flag for overriding the values. Not horrible, but introduces a flag (and only needed for the "classic" FindPythonIntep form).
* We find some way to tell a previous run from a user-set value. This is ideal if I can come up with a way to do it.
* We change scikit-build to include the interpreter path in the cache dir, like setuptools does. Longer term solution, but is a generally good idea, I think - though ideally configurable. More likely to be integrated into scikit-build-core long-term than scikit-build short-term.

**Pull request closed without merging**